include/eve/module/core/regular/impl/simd/x86/convert.hpp

```diff
@@ -10,8 +10,754 @@
+    else  if constexpr( sizeof(T) == 8 && sizeof(U) < 4 )
+    {
+      return convert( convert(v,as<logical<downgrade_t<as_integer_t<T,signed>>>>{}), tgt);
+    }
+    else  if constexpr(   c_in == category::float64x2
+                      && match(c_out, category::int64x2, category::uint64x2)
+                      )
+    {
+      return convert( convert(v,as<logical<float>>{}), tgt);
+    }
+    else  if constexpr( c_in == category::int64x2 )
+    {
+      using i_t = as<logical<std::int32_t>>;
+
+      if constexpr(match(c_out, category::float64x2, category::float32x4))
+        return convert( convert(v,i_t{}), tgt);
+      else
+        return convert_impl(EVE_RETARGET(cpu_),v,tgt);
+    }
+    else if constexpr( c_in == category::int64x4 && c_out == category::float32x4 )
+    {
+      return convert( convert(v,as<logical<std::int32_t>>{}), tgt);
+    }
+    //==============================================================================================
+    // Generic case
+    //==============================================================================================
+    else
+    {
+      return convert_impl(EVE_RETARGET(cpu_),v,tgt);
+    }
+  }
+
+  //================================================================================================
+  // convert: float64 -> U
+  //================================================================================================
+  template<typename N, real_scalar_value U>
+  EVE_FORCEINLINE wide<U, N> convert_impl ( EVE_SUPPORTS(sse2_)
+                                          , wide<double,N> v, as<U> const &tgt
+                                          ) noexcept
+  {
+    constexpr auto c_i = categorize<wide<double, N>>();
+    constexpr auto c_o = categorize<wide<U, N>>();
+
+          if constexpr( c_i == category::float64x2 )
+    {
+            if constexpr( c_o == category::float32x4 ) return _mm_cvtpd_ps(v);
+      else  if constexpr( c_o == category::uint64x2 )
+      {
+        if constexpr(current_api >= avx512) return _mm_cvttpd_epu64(v);
+        else                                return map(convert,v,tgt);
+      }
+      else  if constexpr( c_o == category::int64x2 )
+      {
+        if constexpr(current_api >= avx512) return _mm_cvttpd_epi64(v);
+        else                                return map(convert,v,tgt);
+      }
+      else  if constexpr( match(c_o,category::uint32x4,category::int32x4) )
+      {
+        return _mm_cvttpd_epi32(v);
+      }
+      else  if constexpr( match(c_o,category::uint16x8,category::int16x8) )
+      {
+        return _mm_shufflelo_epi16(_mm_cvttpd_epi32(v), 0b00001000);
+      }
+      else if constexpr( match(c_o,category::uint8x16,category::int8x16) )
+      {
+        return convert( convert(v, as<upgrade_t<U>>{}), tgt);
+      }
+    }
+    else  if constexpr( c_i == category::float64x4 )
+    {
+            if constexpr( c_o == category::float32x4 ) return _mm256_cvtpd_ps(v);
```

Comment:
Can we maybe put the same instructions together? Like

`_mm_cvtpd_ps`, `_mm256_cvtpd_ps`, `_mm512_cvtpd_ps` make sense together